doc/sphinx/proof-engine/vernacular-commands.rst

```diff
@@ -197,24 +188,17 @@ to accessible objects.  (see Section :ref:`invocation-of-tactics`).
       .. insertprodn logical_kind logical_kind
 
       .. prodn::
-         logical_kind ::= @thm_token
-         | @assumption_token
-         | Context
-         | Definition
-         | Example
-         | Coercion
-         | Instance
-         | Scheme
-         | Canonical
-         | Field
-         | Method
-         | Primitive
+         logical_kind ::= {| @thm_token | @assumption_token }
+         | {| Definition | Example | SubClass | Context | Primitive }
+         | {| Coercion | Instance | Scheme }
+         | {| Canonical | Field | Method }
 
       Filters objects by the keyword that was used to define them
       (`Theorem`, `Lemma`, `Axiom`, `Variable`, `Context`,
       `Primitive`...) or its status (`Coercion`, `Instance`, `Scheme`,
       `Canonical`, `Field` for record fields, `Method` for class
-      fields).  Cf. :ref:`this example <search-by-keyword>`.
+      fields).  Note that `Coercion`\s, `Instance`\s and `Scheme`\s can be
+      defined without using those keywords.  See :ref:`this example <search-by-keyword>`.
```

Comment:
According to the code, the previously undocumented `SubClass` is part of the "filter by status" category. So it should be added in the second parenthesis, after `Canonical`, and I suggest moving it to the line below in the `prodn`. I also suggest moving `Canonical` up to the line above, resulting in the following:

```
         logical_kind ::= {| @thm_token | @assumption_token }
         | {| Definition | Example | Context | Primitive }
         | {| Coercion | Instance | Scheme | Canonical | SubClass }
         | {| Field | Method }
```

Canonical Structures can also be defined without using the `Canonical` keyword but the `canonical` attribute instead. I don't know about `SubClass`.